tile/tile_magma.py

```diff
@@ -60,27 +60,48 @@ def __init__(self, core):
             self.wire(self.ports.config.config_data,
                       feature.ports.config.config_data)
             self.wire(self.ports.config.read, feature.ports.config.read)
-            self.wire(self.ports.config.write, feature.ports.config.write)
+            # this is lumped in with config_en logic below
+            # self.wire(self.ports.config.write, feature.ports.config.write)
 
         # read_data mux
         num_mux_inputs = len(self.features())
         self.read_data_mux = MuxWithDefaultWrapper(num_mux_inputs, 32, 8, 0)
-        for i, feat in enumerate(self.features()):
-            self.wire(feat.ports.read_config_data,
-                      self.read_data_mux.ports.I[i])
         # Connect S input to config_addr[feature].
         self.wire(self.ports.config.config_addr[16:24],
                   self.read_data_mux.ports.S)
         self.wire(self.read_data_mux.ports.O, self.ports.read_config_data)
-        self.and2 = FromMagma(mantle.DefineAnd(2))
-        self.eq = FromMagma(mantle.DefineEQ(16))
+        self.read_and_tile = FromMagma(mantle.DefineAnd(2))
+        self.write_and_tile = FromMagma(mantle.DefineAnd(2))
+        self.eq_tile = FromMagma(mantle.DefineEQ(16))
         # config_addr[tile_id] == self.tile_id?
-        self.wire(self.ports.tile_id, self.eq.ports.I0)
-        self.wire(self.ports.config.config_addr[0:16], self.eq.ports.I1)
+        self.wire(self.ports.tile_id, self.eq_tile.ports.I0)
+        self.wire(self.ports.config.config_addr[0:16], self.eq_tile.ports.I1)
         # Connect EN input to (config_addr[tile_id] == self.tile_id & READ)
```

Comment:
should be `config_addr[0:16]`? here and line 76.

Reply:
yeah, I just meant the tile_id field of the config address. I probably shouldn't use that notation.

Maybe `config_addr.tile_id`?

Reply:
got it...that notation seems fine